pkg/doc/verifiable/credential.go

```diff
@@ -0,0 +1,271 @@
+        "type": "string"
+      },
+      "minItems": 2
+    },
+    "credentialSubject": {
+      "anyOf": [
+        {
+          "type": "array"
+        },
+        {
+          "type": "object"
+        }
+      ]
+    },
+    "issuer": {
+      "anyOf": [
+        {
+          "type": "string"
+        },
+        {
+          "type": "object",
+          "required": [
+            "id"
+          ],
+          "properties": {
+            "id": {
+              "type": "string"
+            }
+          }
+        }
+      ]
+    },
+    "issuanceDate": {
+      "$ref": "#/definitions/timestamp"
+    },
+    "proof": {
+      "type": "object",
+      "required": [
+        "type"
+      ],
+      "properties": {
+        "type": {
+          "type": "string"
+        }
+      }
+    },
+    "expirationDate": {
+      "$ref": "#/definitions/timestamp"
+    },
+    "credentialStatus": {
+      "type": "object",
+      "required": ["id", "type"],
+      "properties": {
+        "id": {
+          "type": "string",
+          "format": "uri"
+        },
+        "type": {
+          "type": "string"
+        }
+      }
+    }
+  },
+  "definitions": {
+     "timestamp": {
+      "type": "string",
+      "pattern": "\\d{4}-[01]\\d-[0-3]\\dT[0-2]\\d:[0-5]\\d:[0-5]\\dZ"
+    }
+  }
+}
+`
+
+var schemaLoader = gojsonschema.NewStringLoader(schema)
+
+// Proof defines embedded proof of Verifiable rawCredential
+type Proof struct {
+	Type string `json:"type,omitempty"`
+}
+
+// CredentialStatus defines status of Verifiable rawCredential
+type CredentialStatus struct {
+	ID   string `json:"id,omitempty"`
+	Type string `json:"type,omitempty"`
+}
+
+// Credential Verifiable Credential definition
+type Credential struct {
+	Context    []string
+	ID         string
+	Type       []string
+	Subject    interface{}
+	IssuerID   string
+	IssuerName string
+	Issued     *time.Time
+	Expired    *time.Time
+	Proof      *Proof
+	Status     *CredentialStatus
+}
+
+// rawCredential
+type rawCredential struct {
+	Context []string          `json:"@context,omitempty"`
+	ID      string            `json:"id,omitempty"`
+	Type    []string          `json:"type,omitempty"`
+	Subject interface{}       `json:"credentialSubject,omitempty"`
+	Issued  *time.Time        `json:"issuanceDate,omitempty"`
+	Expired *time.Time        `json:"expirationDate,omitempty"`
+	Proof   *Proof            `json:"proof,omitempty"`
+	Status  *CredentialStatus `json:"credentialStatus,omitempty"`
+	Issuer  interface{}       `json:"issuer,omitempty"`
+}
+
+type issuerPlain struct {
+	ID string `json:"issuer,omitempty"`
+}
+
+type compositeIssuer struct {
+	ID   string `json:"id,omitempty"`
+	Name string `json:"name,omitempty"`
+}
+
+type issuerComposite struct {
+	CompositeIssuer compositeIssuer `json:"issuer,omitempty"`
+}
+
+// FromBytes creates an instance of Verifiable rawCredential by reading a JSON document from bytes
+func FromBytes(data []byte) (*Credential, error) {
```

Comment:
@fqutishat I also see this pattern in doc/did ... Can you remind us why we didn't use use "New" as the method name?

Reply:
I think we should use new I will change doc/did

Reply:
I will change this as well in the verifiable credential.

Reply:
Done